celavi/costgraph.py

```diff
@@ -648,64 +653,91 @@ def choose_paths(self, source_node: str = None, crit: str = "cost"):
                 return _paths
 
     def find_upstream_neighbor(
-        self, node_id: int, crit: str = "dist"
+        self,
+        node_id: int,
+        neighbor_type: List[str] = None,
+        crit: str = 'cost',
     ):
         """
         Given a node in the network, find the "nearest" upstream neighbor to
-        that node that is of the type specified by self.sc_begin. "Nearest" is
-        determined according to the crit parameter.
+        that node that is of the type specified by self.sc_begin or by neighbor_type
```

Comment:
neighbor_type ar elike the sc_in_circ type of edges? The ones that feed into the closed loop?

Reply:
Correct - I made it a parameter rather than hard coding in case we need to use this functionality for another purpose in the future.